### ISSUE TITLE
It can't set option param convenient  when use signal

### ISSUE THREAD

**Issue by zszen** [godotengine/godot#32155]
**Godot version:**
3.1.1

**OS/device including version:**
mac 10.14.4

**Issue description:**
If create signal with param, set null or empty it can't be post success.

**Steps to reproduce:**
1. create a signal 
```
signal deal(p1,p2)
```
2. connec it
```
sth.connect("deal",self,"on_deal")
```
3. post event
>fail1
```
sth.emit_signal("deal", p1, null)
```
>fail2
```
sth.emit_signal("deal", p1)
```
>success
```
sth.emit_signal("deal", p1, {})
```

>Is it that for support optional param, I need create a new Object fill it. Waste mem for this.

**Minimal reproduction project:**

**Comment by zszen** [comment]
And I found that. If connect function param defined type, the emit_signal post with difference type param. It nothing happend. 

Could the godot support warning for this condition ?

**Comment by KoBeWi** [comment]
Fixed in 4.0
